searcher/typesense/client.go

```diff
@@ -0,0 +1,97 @@
+package typesense
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/util"
+	"github.com/typesense/typesense-go/typesense"
+	"github.com/typesense/typesense-go/typesense/api"
+)
+
+type Typesense struct {
+	client *typesense.Client
+}
+
+func NewTypesenseClient(searchConfig config.SearchConfiguration) (*Typesense, error) {
+	client := typesense.NewClient(
+		typesense.WithServer(searchConfig.Typesense.Host),
+		typesense.WithAPIKey(searchConfig.Typesense.ApiKey),
+	)
+
+	_, err := client.Health(5 * time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Typesense{client: client}, err
+}
+
+func (t *Typesense) Search(f *datastore.Filter) ([]string, datastore.PaginationData, error) {
+	events := make([]string, 0)
+	data := datastore.PaginationData{}
+	queryByBuilder := new(strings.Builder)
+	filterByBuilder := new(strings.Builder)
+
+	filterByBuilder.WriteString(fmt.Sprintf("app_metadata.group_id:=%s", f.Group.UID))
+
+	hasAppFilter := !util.IsStringEmpty(f.AppID)
+	if hasAppFilter {
+		filterByBuilder.WriteString(fmt.Sprintf(" && app_metadata.uid:=%s", f.Group.UID))
+	}
+
+	filterByBuilder.WriteString(fmt.Sprintf(" && created_at:[%d..%d]", f.SearchParams.CreatedAtStart*1000, f.SearchParams.CreatedAtEnd*1000))
+
```

Comment:
Hm, i'm confused, why multiply by 1000?

Reply:
Because the search middleware converts epoch in seconds but we need them in milliseconds

Reply:
i see.